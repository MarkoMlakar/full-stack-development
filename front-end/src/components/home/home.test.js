import React from "react";
import { render, waitFor, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Home from "./home";

afterEach(cleanup);

test("renders correctly", () => {
  const tree = render(<Home />);
  expect(tree).toMatchSnapshot();
});
