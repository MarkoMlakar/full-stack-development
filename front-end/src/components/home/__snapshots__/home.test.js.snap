// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <div
          class="top-section"
        >
          <div
            class="headline"
          >
            <div
              data-testid="home-title"
            >
              Discover flowers around you
            </div>
            <div
              class="subtitle"
            >
              Explore between more than 8.427 sightings
            </div>
            <div
              class="search-bar"
            >
              <input
                class="input-field"
                placeholder="Looking for something specific?"
              />
              <img
                alt="search-bar"
                src="pl-icon-search.svg"
              />
            </div>
          </div>
        </div>
        <div
          class="bottom-section"
        >
          <div
            class="home-spinner"
          >
            <img
              alt="Loading..."
              class="rotate"
              src="flower-logo.svg"
            />
            ;
          </div>
          <div
            class="image-list"
            data-testid="image-list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <div
        class="top-section"
      >
        <div
          class="headline"
        >
          <div
            data-testid="home-title"
          >
            Discover flowers around you
          </div>
          <div
            class="subtitle"
          >
            Explore between more than 8.427 sightings
          </div>
          <div
            class="search-bar"
          >
            <input
              class="input-field"
              placeholder="Looking for something specific?"
            />
            <img
              alt="search-bar"
              src="pl-icon-search.svg"
            />
          </div>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <div
          class="home-spinner"
        >
          <img
            alt="Loading..."
            class="rotate"
            src="flower-logo.svg"
          />
          ;
        </div>
        <div
          class="image-list"
          data-testid="image-list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
